### ISSUE TITLE
Cannot remove data object with single quotation mark in name

### ISSUE THREAD

**Issue by ll4strw** [irods/irods_rule_engine_plugin_logical_quotas#94]
- [x] main
- [x] 4-3-stable

## Bug Report

- irods v4.3.0
- OS Rocky v8
- icommands v4.3.0
- irods-rule-engine-plugin-logical-quotas v4.3.0

Start monitoring a collection using the quota  REP,  for instance 

```
imeta ls -C .
AVUs defined for collection /zone/home/testuser:

----
attribute: irods::logical_quotas::maximum_size_in_bytes
value: 21474836480
units: 
----
attribute: irods::logical_quotas::total_number_of_data_objects
value: 6
units: 
----
attribute: irods::logical_quotas::total_size_in_bytes
value: 99079
units: 

```

then try adding to the monitored collection a data object whose name contains a single quotation mark.

``iput test1 "just testin'"`` completes without errors and the file test1 is successfully transferred on the server. However, ``irm -f "just testin'"`` returns

```
remote addresses: x.x.x.x ERROR: rmUtil: rm error for /zone/home/testuser/just testin', status = -1205000 status = -1205000 RE_RUNTIME_ERROR
Level 0: no good replica found: Unknown error -166000
```

_Originally reported at https://github.com/irods/irods/issues/7164._

**Comment by trel** [comment]
`ils` of the data object after `iput` shows no permissions on the data object.

**Comment by trel** [comment]
> This means that there is also an issue with how iRODS handles objects names?

I ... don't think so?   Or at least I haven't seen that yet.

I thought the missing permissions were only happening when this REP was present...
(without the Logical Quotas being installed/activated... there was no issue).

**Comment by korydraughn** [comment]
I thought that as well.

I never saw this issue on a fresh install of 4.3.0.

**Comment by trel** [comment]
> If I type ils -lA, nota bene without an input file name, the resulting list shows the right ACLs including for the just testin' object.

oh!   that *is* curious/new.

**Comment by korydraughn** [comment]
Everything worked for me on a fresh install. This was run on Ub20 with Postgres 10.
```
root@cc9948e56928:/# su - irods                                            
irods@cc9948e56928:~$ ils                                                  
/tempZone/home/rods:                                                       
irods@cc9948e56928:~$ cat version.json                                     
{                                                                          
    "catalog_schema_version": 9,                                           
    "commit_id": "f6eb6c72786288878706e2562a370b91b7d0802e",               
    "installation_time": "2023-07-05T11:43:52.776433",                     
    "irods_version": "4.3.0",                                              
    "schema_name": "version",                                              
    "schema_version": "v4"                                                 
}irods@cc9948e56928:~$ iput version.json "just testin'"                    
irods@cc9948e56928:~$ ils -l                                               
/tempZone/home/rods:                                                       
  rods              0 demoResc          239 2023-07-05.11:46 & just testin'
irods@cc9948e56928:~$ ils -A                                               
/tempZone/home/rods:                                                       
        ACL - rods#tempZone:own                                            
        Inheritance - Disabled                                             
  just testin'                                                             
        ACL - rods#tempZone:own                                            
irods@cc9948e56928:~$ ils -lA                                              
/tempZone/home/rods:                                                       
        ACL - rods#tempZone:own                                            
        Inheritance - Disabled                                             
  rods              0 demoResc          239 2023-07-05.11:46 & just testin'
        ACL - rods#tempZone:own                                            
irods@cc9948e56928:~$ ils -lA "just testin'"                               
  rods              0 demoResc          239 2023-07-05.11:46 & just testin'
        ACL - rods#tempZone:own                                            
```

**Comment by korydraughn** [comment]
Oh no! Reproduced it!
```
irods@cc9948e56928:~$ ils -A "just testin'"
  /tempZone/home/rods/just testin'         
        ACL -                              
```

**Comment by korydraughn** [comment]
That may be true.

I'll verify and depending on what happens, we can repurpose this issue for the UNLINK case.

**Comment by korydraughn** [comment]
I've confirmed that the unlink issue is indeed due to the REP.

**Comment by korydraughn** [comment]
I've looked at this for the upcoming 4.3.2 release and here's what I've found.

PR #110 resolves the removal/unlink issue by allowing the operation to continue if the filesystem library throws `SYS_NO_GOOD_REPLICAS`
- See the following lines in the filesystem library.
    - https://github.com/irods/irods/blob/e4b4e8a7266360bb38ee97d7f2eda69249567dff/lib/filesystem/src/filesystem.cpp#L761-L763
- Data objects containing single quotes will result in `SYS_NO_GOOD_REPLICAS` being thrown
- The outcome of this is the data object will be removed, but the quota for total bytes will not be updated. Admins will need to recalculate the totals, which is okay since there is a workaround
    - Workaround: Add a specific query or switch to GenQuery2 for the lookup

Given there's a way to workaround the quota update issue, we'll handle that after 4.3.2 is released. I'll open an issue pointing to this comment so we don't forget about it.
